share/zfsnap/commands/send.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/sh
+
+# This file is licensed under the BSD-3-Clause license.
+# See the AUTHORS and LICENSE files for more information.
+
+SSH_KEY=~/.ssh/id_rsa
+#EMAIL=none@none
+user=$USER
+LOCAL='false'
+FULL='false'
+REVERSE='false'
+
+# FUNCTIONS
+Help() {
+    cat << EOF
+${0##*/} v${VERSION}
+
+Syntax:
+${0##*/} destroy [ options ] zpool/filesystem ...
+
+OPTIONS:
+  -b           = source (to be backed up) zfs pool and filesystem, pool_name/filesystem_name
+  -d           = destination (to store backup) zfs pool and filesystem, pool_name/filesystem_name
+  #-e		   = email address to send notification to
+  #-f 		   = enable sendin full backup rather than incremental
+  -h           = Print this help and exit
+  -i		   = remote zfs systems IP address or DNS name
+  #-l		   = indicates this will be a backup to a local zfs file system and ssh is not needed
+  -k 		   = ssh key to be used to authenticate to remote zfs system; default ~/.ssh/id_rsa
+  -n           = Dry-run. Perform a trial run with no backup actually performed
+  -R		   = source zfs system ssh's to destination and sends snapshot instead of default destination zfs sytem ssh'ing to source and telling it to send the snapshot back
+  -s           = Skip pools that are resilvering
+  -S           = Skip pools that are scrubbing
+  -u 		   = user for authentication via ssh to remote zfs system; default current user
+  -v           = Verbose output
+
+LINKS:
+  website:          http://www.zfsnap.org
+  repository:       https://github.com/zfsnap/zfsnap
+  bug tracking:     https://github.com/zfsnap/zfsnap/issues
+
+EOF
+    Exit 0
+}
+
+# main loop; get options, perform backup of filesystem(s)
+while [ -n "$1" ]; do
+    OPTIND=1
+    while getopts b:d:hi:k:nRsSu:v OPT; do
+        case "$OPT" in
+            b) SOURCEFS=$OPTARG;;
+            d) DESTINATIONFS=$OPTARG;;
+            #e) EMAIL=$OPTARG;;#could possibly have utility email results of backup upon completion, would require user to already have setup mail
+            #f) FULL='TRUE';;#this flag can be used to perform a full backup when full backup option is implemented
+            h) Help;;
+			i) REMOTE=$OPTARG;;#remote system ip or dns address
+            k) SSH_KEY=$OPTARG;;#ssh key to access remote system
+			#l) LOCAL='true';;#enables backing up to a local filesystem
+            n) DRY_RUN='true';;
+            R) REVERSE='true';;
+            s) PopulateSkipPools 'resilver';;
+            S) PopulateSkipPools 'scrub';;
+			u) user=$OPTARG;;
+            v) VERBOSE='true';;
+
+            :) Fatal "Option -${OPTARG} requires an argument.";;
+           \?) Fatal "Invalid option: -${OPTARG}.";;
+        esac
+    done
+
+    # discard all arguments processed thus far
+    shift $(($OPTIND - 1))
+	
+	#test to see if SSH_KEY specified exists as long as this is not a local backup
+	if ! $LOCAL; then
+		if [ ! -r $SSH_KEY ]; then
```

Comment:
$SSH_KEY should be double quoted here, to protect against a space being in the path name.